servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/ErrorClass.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+/**
+ * Enumeration of the main failure classes.
+ */
+enum ErrorClass {
```

Comment:
Are these known conditions, that the LB can map to, or are we expecting a translation layer within the LB (user influenced?). 
On a side note, would we have a way to create a custom request tracker through the xDS protocol, and map these to penalties?